File: libgomp/testsuite/libgomp.c/examples-4/array_sections-4.c
/* { dg-do run } */

#include <stdlib.h>

void foo ()
{
  int A[30], *p;
  #pragma omp target data map(A[0:10])
    {
      p = &A[0];
      #pragma omp target map(p[3:7]) map(A[0:10])
	{
	  A[2] = 777;
	  A[8] = 777;
	  p[8] = 999;
	}
    }

  if (A[2] != 777 || A[8] != 999)
    abort ();
}

int main ()
{
  foo ();
  return 0;
}

File: libgomp/testsuite/libgomp.fortran/examples-4/array_sections-4.f90
! { dg-do run }

  call foo ()
contains
  subroutine foo ()
    integer, target :: A(30)
    integer, pointer :: p(:)
    !$omp target data map(A(1:10))
      p => A
      !$omp target map(p(4:10)) map(A(1:10))
        A(3) = 777
        p(9) = 777
        A(9) = 999
      !$omp end target
    !$omp end target data
    if (A(3) /= 777 .or. A(9) /= 999) stop 1
  end subroutine
end
